reflex/state.py

```diff
@@ -738,13 +738,43 @@ async def _process(self, event: Event) -> AsyncIterator[StateUpdate]:
             # Clean the state to prepare for the next event.
             self._clean()
 
+    def _check_valid(self, handler: EventHandler, events: Any) -> Any:
+        """Check if the events yielded are valid. They must be EventHandlers or EventSpecs.
+
+        Args:
+            handler: EventHandler.
+            events: The events to be checked.
+
+        Raises:
+            TypeError: If any of the events are not valid.
+
+        Returns:
+            The events as they are if valid.
+        """
+        if (
+            events is None
+            or (
+                isinstance(events, list)
+                and all(isinstance(e, (EventHandler, EventSpec)) for e in events)
+            )
+            or (
+                not isinstance(events, list)
+                and isinstance(events, (EventHandler, EventSpec))
+            )
+        ):
+            return events
+
+        raise TypeError(
+            f"Your handler {handler.fn.__qualname__} must only return/yield: None, Events or other EventHandlers"
```

Comment:
```suggestion
            f"Your handler {handler.fn.__qualname__} must only return/yield: None, Events or other EventHandlers referenced by their class (not using `self`)"
```

maybe give a bit more hint to users about how to reference other handlers.